src/lib/y2storage/storage_class_wrapper.rb

```diff
@@ -0,0 +1,258 @@
+      # @param as [String] name of the class to convert the result, nil if the
+      #   value must be returned as-is (after turning any vector into an array)
+      # @param raise_errors [Array<String>] name of the exceptions that must not
+      #   be turned into a nil result. Can contain "WrongNumberOfChildren" and/or
+      #   "DeviceHasWrongType". Useful for methods in which those exceptions
+      #   don't have the usual meaning (looking for something that is not
+      #   there).
+      #
+      # @example Usage of storage_forward
+      #
+      #   module Y2Storage
+      #     class BlkDevice < Device
+      #       include StorageClassWrapper
+      #       wrap_class Storage::BlkDevice
+      #
+      #       storage_forward :name
+      #       storage_forward :size, as: "DiskSize"
+      #       storage_forward :size=
+      #       storage_forward :udev_paths
+      #       storage_forward :blk_filesystem, as: "Filesystems::BlkFilesystem"
+      #       storage_forward :create_blk_filesystem,
+      #         as: "Filesystems::BlkFilesystem",
+      #         raise_errors: ["WrongNumberOfChildren"]
+      #     end
+      #   end
+      #
+      #   device = Y2Storage::BlkDevice.new(storage_device)
+      #
+      #   # This is simply forwarded
+      #   device.name
+      #
+      #   # Returns a Y2Storage::DiskSize object instead of the Fixnum (bytes)
+      #   # returned by Storage::BlkDevice#size
+      #   device.size
+      #
+      #   # Transforms the argument to Fixnum (bytes) before passing it to
+      #   # Storage::BlkDevice#size=
+      #   device.size = Y2Storage::DiskSize.GiB(20)
+      #
+      #   # Returns an array of strings instead of a Storage::VectorString
+      #   device.udev_paths
+      #
+      #   # If there is no filesystem (device is not formatted), it returns
+      #   # nil instead of raising an exception. If there is one, its wrapped
+      #   # into a Y2Storage::Filesystems::BlkFilesystem object before being
+      #   # returned.
+      #   device.blk_filesystem
+      #
+      #   fs_type = Y2Storage::Filesystems::Type.find(:btrfs)
+      #   # Transforms fs_type into something understandable by
+      #   # Storage::BlkDevice#create_blk_device when passing it.
+      #   # If the device was already formatted, it raises an exception
+      #   # (is not masked like in the example above).
+      #   device.create_blk_filesystem(fs_type)
+      #
+      def storage_forward(method, as: nil, raise_errors: [])
+        modifiers = { as: as, raise_errors: raise_errors }
+        define_method(method) do |*args|
+          StorageClassWrapper.forward(to_storage_value, method, modifiers, *args)
+        end
+      end
+
+      # Equivalent to #storage_forward used to define class methods instead of
+      # instance ones.
+      #
+      # @see #storage_forward
+      def storage_class_forward(method, as: nil, raise_errors: [])
+        modifiers = { as: as, raise_errors: raise_errors }
```

Comment:
It's for the sake of better autogenerated Yardoc documentation. Key attributes are way more clear (with their default value and everything) that hashes that must be described with text.